sim/core/debuffs.go

```diff
@@ -453,26 +455,71 @@ func ScheduledSunderArmorAura(target *Unit) *Aura {
 	return aura
 }
 
-func ExposeArmorAura(target *Unit, talentPoints int32) *Aura {
-	armorReduction := 2050.0 * (1.0 + 0.25*float64(talentPoints))
+var AcidSpitAuraLabel = "Acid Spit"
+
+func AcidSpitAura(target *Unit, startingStacks int32) *Aura {
+	armorReductionPerStack := 0.1
 
 	return target.GetOrRegisterAura(Aura{
-		Label:    "ExposeArmor-" + strconv.Itoa(int(talentPoints)),
-		Tag:      SunderExposeAuraTag,
-		ActionID: ActionID{SpellID: 26866},
-		Duration: time.Second * 30,
+		Label:     AcidSpitAuraLabel,
+		Tag:       MajorArmorReductionTag,
+		ActionID:  ActionID{SpellID: 55745}, // TODO: Spell id
+		Duration:  time.Second * 10,
+		MaxStacks: 2,
+		Priority:  armorReductionPerStack * 2,
+		OnGain: func(aura *Aura, sim *Simulation) {
+			aura.SetStacks(sim, startingStacks)
+		},
+		OnStacksChange: func(aura *Aura, sim *Simulation, oldStacks int32, newStacks int32) {
+			oldMultiplier := (1.0 - float64(oldStacks)*armorReductionPerStack)
+			newMultiplier := (1.0 - float64(newStacks)*armorReductionPerStack)
+			aura.Unit.PseudoStats.ArmorMultiplier *= (newMultiplier / oldMultiplier)
+		},
+	})
+}
+
+func ScheduledAcidSpitAura(target *Unit) *Aura {
```

Comment:
Instead of defining a new function, lets make a ScheduledAura() function which takes additional parameters for the aura and time to apply.

Reply:
Made this change but may need some tweaks. I'll leave the comment unresolved until I get some feedback on it. Specifically, not sure whether it makes sense to use the options structure as a parameter (as I have currently). Primary reason I chose to was that the OnAction handler is different but that difference seems like it could be inferred (in all of the current use cases).